src/main/java/nl/tudelft/contextproject/gui/CreateScriptController.java

```diff
@@ -347,60 +598,92 @@ private void setBackButton() {
                 }
             }
 
+            Script bak = new Script(backupList);
+            bak.setName(ContextTFP.getScript().getName());
+            
+            ContextTFP.setScript(bak);
             MenuController.show();
         });
     }
-
+    
     /**
-     * Sets the onAction for the save button.
+     * Sets the onAction for the save buttons.
      */
     private void setSaveButton() {
+        if (backupList.isEmpty()) {
+            btnSave.setDisable(true);
+        }
+        
         btnSave.setOnAction(event -> {
+            setSavePopup(event, false);
+        });
+        
+        btnSaveAs.setOnAction(event -> {
+            setSavePopup(event, true);
+        });
+    }
+
+    /**
+     * Handles the saving of a file.
+     */
+    private void setSavePopup(ActionEvent event, boolean showDialog) {
+        final Script script = new Script(tableEvents.getItems());
+
+        if (!script.showValid(1)) {
+            return;
+        }
+        
+        File file;
+        
+        if (showDialog) {
             FileChooser fileChooser = new FileChooser();
             fileChooser.setTitle("Save script");
             fileChooser.setInitialFileName("script");
             fileChooser.getExtensionFilters().add(new ExtensionFilter("XML (*.xml)", "*.xml"));
+            
+            file = fileChooser.showSaveDialog(((Node) event.getTarget()).getScene().getWindow());
+        } else {
+            file = new File(SaveScript.getSaveLocation());
+        }
 
-            File file = fileChooser.showSaveDialog(((Node) event.getTarget()).getScene().getWindow());
-
-            if (file != null) {
-                try {
-                    SaveScript.setSaveLocation(file.getAbsolutePath());
-                    SaveScript.save(script);
+        if (file != null) {
+            try {
+                SaveScript.setSaveLocation(file.getAbsolutePath());
+                SaveScript.save(script);
 
-                    script.setName(file.getName());
-                    ContextTFP.setScript(script);
+                script.setName(file.getName());
+                ContextTFP.setScript(script);
 
-                    Alert alert = new Alert(AlertType.CONFIRMATION);
-                    alert.setTitle("Confirm exiting");
-                    alert.setHeaderText("Saving script was succesful!");
-                    alert.setContentText("Succesful save of script: " 
-                            + file.getName()
-                            + " Do you want to quit to menu?");
+                Alert alert = new Alert(AlertType.CONFIRMATION);
+                alert.setTitle("Confirm exiting");
+                alert.setHeaderText("Saving script was succesful!");
+                alert.setContentText("Succesful save of script: " 
+                        + file.getName()
+                        + " Do you want to quit to menu?");
```

Comment:
Maybe you can put this alert in a new method to reduce the length of this method.